File: Praktikum-4/Php/Dosen.php
<?php

class Dosen extends PegawaiTetap
{
    private float $tunjanganFungsional;

    public function __construct(
        string $nip,
        string $nama,
        float $gajiPokok,
        int $masaKerja,
        float $tunjanganFungsional
    ) {
        parent::__construct($nip, $nama, $gajiPokok, $masaKerja);

        $this->tunjanganFungsional = $tunjanganFungsional;
    }

    public function hitungGaji(): float
    {
        return parent::hitungGaji() + $this->tunjanganFungsional;
    }

    public function jenis(): string
    {
        return "DOSEN";
    }

    protected function getMasaKerjaTahunan(): int
    {
        return parent::getMasaKerjaTahun();
    }
}

File: Praktikum-4/Java/Dosen.java
public class Dosen extends PegawaiTetap {
    private double tunjanganFungsional;

    public Dosen(String nip, String nama, double gajiPokok, int masaKerja, double tunjanganFungsional) {
        super(nip, nama, gajiPokok, masaKerja);

        this.tunjanganFungsional = tunjanganFungsional;
    }


    @Override
    public double hitungGaji(){
        return super.hitungGaji() + this.tunjanganFungsional;
    }

    @Override
    public String jenis(){
        return "DOSEN";
    }

    protected int getMasaKerjaTahunan(){
        return super.getMasaKerjaTahun();
    }
}